src/App.tsx

```diff
@@ -15,36 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_ALPH = 'alphabetical';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods: string[],
+  { sortType, isReverse }: { sortType: string; isReverse: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SORT_FIELD_ALPH:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const handleReset = () => {
+    setSortField('');
+    setIsReverse(false);
+  };
+
+  const handleSortByAlphabet = () => {
+    setSortField(SORT_FIELD_ALPH);
+  };
+
+  const handleSortByLength = () => {
+    setSortField(SORT_FIELD_LENGTH);
+  };
+
+  const handleReverse = () => {
+    setIsReverse(prevState => !prevState);
```

Comment:
The function's name should start with a verb. Consider renaming `handleReverse` to `toggleReverse` to better describe the action it performs.